adapters/repos/db/index.go

```diff
@@ -320,41 +320,36 @@ func indexID(class schema.ClassName) string {
 	return strings.ToLower(string(class))
 }
 
-func (i *Index) shardFromUUID(in strfmt.UUID, ss *sharding.State) (string, error) {
-	uuid, err := uuid.Parse(in.String())
-	if err != nil {
-		return "", errors.Wrap(err, "parse id as uuid")
-	}
-
-	uuidBytes, _ := uuid.MarshalBinary() // cannot error
-
-	return ss.PhysicalShard(uuidBytes), nil
-}
-
 func (i *Index) determineObjectShard(id strfmt.UUID, tenantKey string, ss *sharding.State) (string, error) {
```

Comment:
can we break this function up in a "getShardingState" function and then call `shardFromTenantKey` and `shardFromUUID` directly? I noticed that some callers of this function already have a `if tenant != ""` around it and that makes the flow somewhat confusing

Reply:
Don't like it either. Also to be refactored.